pandas/src/period.pyx

```diff
@@ -458,13 +458,39 @@ def extract_ordinals(ndarray[object] values, freq):
 
     for i in range(n):
         p = values[i]
-        ordinals[i] = p.ordinal
-        if p.freqstr != freqstr:
-            msg = _DIFFERENT_FREQ_INDEX.format(freqstr, p.freqstr)
-            raise IncompatibleFrequency(msg)
+
+        if p is tslib.NaT:
+            ordinals[i] = tslib.iNaT
```

Comment:
are these *only* integers at this point? e.g. we normally use a very generic check for NaT, ``is_null_datetimelike``